### PR TITLE
band aids to make the API a bit less destructive

### PR DESCRIPTION
**Pull request by chris48s**
This just fixes the serious bugs. There is still some usability stuff to address.
Refs #124

### CODE REVIEW

**Comment by chris48s** [comment]
In the spirit of this becoming a bit more of a community-driven project, I'm happy for @boffbowsh or @kirknorthrop to take the initiative on reviewing this - I think you've both got permission to merge PRs on this repo

File: electionleaflets/apps/api/views.py [hunk 1]
```diff
@@ -24,6 +25,17 @@ class StandardResultsSetPagination(LimitOffsetPagination):
...
 
+class LeafletPermissions(BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        # Allow unauthenticated users to GET and POST
+        # but not PUT, PATCH and DELETE
+        if request.method in ['GET', 'POST']:
+            return True
+        else:
+            return request.user == obj
```
Comment (boffbowsh) [inline electionleaflets/apps/api/views.py]: Should this be `return request.user == obj.owner`?

File: electionleaflets/apps/api/tests/test_create_leaflet.py [hunk 2]
```diff
@@ -21,4 +21,4 @@ def test_create_leaflet(self):
         response = self.client.post(leaflet_url, {}, format='json')
         self.assertEqual(response.data['status'], 'draft')
         leaflet_id = response.data['pk']
-        self.assertEqual(leaflet_id, 1)
+        self.assertTrue(leaflet_id > 0)
```
Comment (boffbowsh) [inline electionleaflets/apps/api/tests/test_create_leaflet.py]: I think using `assertGreater` here would give a nicer error message

File: electionleaflets/apps/api/views.py (hunk 1 above)
Comment (chris48s) [inline electionleaflets/apps/api/views.py]: good spot

**Comment by chris48s** [comment]
both fixed- ta

**Pull request merged**